plasmapy/dispersion/numerical/hollweg_.py

```diff
@@ -0,0 +1,339 @@
+
+    # validate z_mean
+    if z_mean is None:
+        try:
+            z_mean = abs(ion.integer_charge)
+        except ChargeError:
+            z_mean = 1
+    else:
+        if not isinstance(z_mean, (int, np.integer, float, np.floating)):
+            raise TypeError(
+                f"Expected int or float for argument 'z_mean', but got {type(z_mean)}."
+            )
+        z_mean = abs(z_mean)
+
+    # validate arguments
+    for arg_name in ("B", "n_i", "T_e", "T_i"):
+        val = locals()[arg_name].squeeze()
+        if val.shape != ():
+            raise ValueError(
+                f"Argument '{arg_name}' must be single valued and not an array of "
+                f"shape {val.shape}."
+            )
+        locals()[arg_name] = val
+
+    # validate arguments
+    for arg_name in ("gamma_e", "gamma_i"):
+        if not isinstance(locals()[arg_name], (int, np.integer, float, np.floating)):
+            raise TypeError(
+                f"Expected int or float for argument '{arg_name}', but got "
+                f"{type(locals()[arg_name])}."
+            )
+
+    # validate argument k
+    k = k.squeeze()
+    if not (k.ndim == 0 or k.ndim == 1):
+        raise ValueError(
+            f"Argument 'k' needs to be single valued or a 1D array astropy Quantity,"
+            f" got array of shape {k.shape}."
+        )
+    if np.any(k <= 0):
+        raise ValueError("Argument 'k' can not be a or have negative values.")
+
+    # validate argument theta
+    theta = theta.squeeze()
+    if not (theta.ndim == 0 or theta.ndim == 1):
+        raise ValueError(
+            f"Argument 'theta' needs to be a single valued or 1D array astropy "
+            f"Quantity, got array of shape {k.shape}."
+        )
+
+    # Single k value case
+    if np.isscalar(k.value):
+        k = np.array([k.value,]) * u.rad / u.m
+
+    # Calc needed plasma parameters
+    n_e = z_mean * n_i
+    c_s = pfp.ion_sound_speed(
+        T_e=T_e,
+        T_i=T_i,
+        ion=ion,
+        n_e=n_e,
+        gamma_e=gamma_e,
+        gamma_i=gamma_i,
+        z_mean=z_mean,
+    )
+    v_A = pfp.Alfven_speed(B, n_i, ion=ion, z_mean=z_mean)
+    omega_ci = pfp.gyrofrequency(B=B, particle=ion, signed=False, Z=z_mean)
+    omega_pe = pfp.plasma_frequency(n=n_e, particle="e-")
+    cs_vA = c_s / v_A
+
+    # Parameters kx and kz
+    kz = np.cos(theta.value) * k
+    kx = np.sqrt(k ** 2 - kz ** 2)
+
+    # Bellan2012JGR beta param equation 3
+    beta = (c_s / v_A) ** 2
+
+    # Parameters D, F, sigma, and alpha to simplify equation 3
+    D = (c_s / omega_ci) ** 2
+    F = (c / omega_pe) ** 2
+    sigma = (kz * v_A) ** 2
+    alpha = (k * v_A) ** 2
+
+    # Polynomial coefficients: c3*x^3 + c2*x^2 + c1*x + c0 = 0
+    c3 = F * kx ** 2 + 1
+    c2 = -sigma * ((alpha / sigma) * (1 + beta + F * kx ** 2) + D * kx ** 2 + 1)
+    c1 = sigma * alpha * (1 + 2 * beta + D * kx ** 2)
+    c0 = -beta * alpha * sigma ** 2
```

Comment:
```suggestion
    c3 = F * kx ** 2 + 1
    c2 = -alpha_A * (1 + beta + F * kx ** 2) - sigma * (1 + D * kx ** 2)
    c1 = sigma * alpha_A * (1 + 2 * beta + D * kx ** 2)
    c0 = -alpha_s * sigma ** 2
```

Applying changes from https://github.com/PlasmaPy/PlasmaPy/pull/1189/files#r767014478